types/interface-connection/index.d.ts

```diff
@@ -5,9 +5,12 @@
 
 /// <reference types="multiaddr"/>
 
+// model after abortable-iterator
+type Source<T> = AsyncIterable<T> | Iterable<T>;
+type Sink<TSource, TReturn = void> = (source: Source<TSource>) => TReturn
 declare interface Stream {
```

Comment:
```suggestion
declare interface Stream<TSource = unknown, TReturn = unknown> {
```

and use thos on sink and source methods :slightly_smiling_face: